File: 2021--白冠鸡优化算法（Coot optimization algorithm，COOT）/MATLAB/main.m
% Developed in MATLAB R2017b
% Source codes demo version 1.0
% _____________________________________________________

%  Author, inventor and programmer: Iraj Naruei and Farshid Keynia,
%  e-Mail: irajnaruei@iauk.ac.ir , irajnaruei@yahoo.com
% _____________________________________________________
%  Co-author and Advisor: Farshid Keynia
%
%         e-Mail: fkeynia@gmail.com
% _____________________________________________________
%  Co-authors: Amir Sabbagh Molahoseyni
%        
%         e-Mail: sabbagh@iauk.ac.ir
% _____________________________________________________

% You can find the COOT code at 
% _____________________________________________________
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

clear all 
close all
clc

N=30; % Number of search agents

Function_name='F13'; % Name of the test function 

Max_iter=500; % Maximum number of iterations

% Load details of the selected benchmark function
[lb,ub,dim,fobj]=Get_Functions_details(Function_name);

[Convergence_curve,gBest,gBestScore]=COOT(N,Max_iter,lb,ub,dim,fobj);

display(['The best location of COOT is: ', num2str(gBest)]);
display(['The best fitness of COOT is: ', num2str(gBestScore)]);

        

File: 2021--白冠鸡优化算法（Coot optimization algorithm，COOT）/PYTHON/main.py
from Fuction import  Get_Functions_details,COOT
from matplotlib import pyplot as plt

N=30                                # Number of search agents

Function_name='F13' # Name of the test function 

Max_iter=500                # Maximum number of iterations

# Load details of the selected benchmark function
lb,ub,dim,fobj =Get_Functions_details(Function_name)

[Convergence_curve,gBest,gBestScore]=COOT(N,Max_iter,lb,ub,dim,fobj)

print('The best location of COOT is: ', gBest)
print('The best fitness of COOT is: ', gBestScore)

plt.rcParams['font.sans-serif'] = ['SimHei'] # 解决汉字显示为□指定默认字体为黑体。
plt.rcParams['axes.unicode_minus'] = False   # 解决保存图像时 负号'-' 显示为□和报错的问题
#Draw objective space
# plt.subplot(1,2,2)
plt.plot(range(len(Convergence_curve.T)),Convergence_curve.T)

plt.title('Objective space')
plt.xlabel('Iterations')
plt.ylabel('Best score')
plt.grid()
plt.title('COOT')
plt.show()